### PR TITLE
VB-570: Add test for prisoner profile page

### PR DESCRIPTION
**Pull request by psoleckimoj**
Lots of changes:

- Updated types to allow data attributes for testing
- Added integration tests for prisoner profile
- Refactored test stubs to allow data to be passed in
- Lots more!!

### CODE REVIEW

File: server/services/prisonerProfileService.ts [hunk 1]
```diff
@@ -152,31 +187,55 @@ export default class PrisonerProfileService {
...
-          { text: `${properCase(visit.visitStatus)}` },
         ] as PastVisitItem
       })
     )
 
     return visitsForDisplay
   }
 
-  private async getPrisonerSocialContacts(contacts: Contact[], contactIds: number[]): Promise<string[]> {
+  private getPrisonerSocialContacts(contacts: Contact[], visitors: Visitor[]): string[] {
```
Comment (psoleckimoj) [inline server/services/prisonerProfileService.ts:232]: Not strictly needed but spotted it whilst doing other bits, removes more duplication.

File: server/utils/utils.ts [hunk 2]
```diff
@@ -109,14 +109,18 @@ export const convertToTitleCase = (sentence: string): string =>
 export const visitDateAndTime = ({
   startTimestamp,
   endTimestamp,
+  isIntTest = false,
```
Comment (psoleckimoj) [inline server/utils/utils.ts:112]: Made this an optional, allows for the integration tests to return a different format for checking as cypress removes HTML tags.

File: server/services/prisonerProfileService.ts (hunk 1 above)
Comment (psoleckimoj) [reply server/services/prisonerProfileService.ts:232]: Also realised it wasn't actually an async function.

**Review by tpmcgowan — APPROVED** [review]
LGTM!

**Pull request merged**